### ISSUE TITLE
Problems with unicode and striptags filter

### ISSUE THREAD

**Issue by brigadier** [erlydtl/erlydtl#205]
Erlang 17.5
Problem 1:
```
> S = unicode:characters_to_binary("абвé", unicode, utf8).
<<208,176,208,177,208,178,195,169>>

> {ok, tmpl} = erlydtl:compile_template("{{ s|striptags }}", tmpl).
{ok,tmpl}

> tmpl:render([{s, S}]). 
* exception error: bad argument
     in function  re:replace/4
        called as re:replace([1072,1073,1074,233],"(<[^>]+>)",[],[global,{return,list}])
     in call from tmpl:render_internal/2 (, line 1121)
     in call from tmpl:render/2 (, line 495)
```

`replace` function should have `unicode` and `ucp` flags in options. 

Problem 2:
Sometimes, depending on input data, this filer still works, but it always returns result in `latin1` encoding instead of much more common `utf-8`. As a result, the page which contains variables with this filter, shows in browser with unicode errors (question signs) in content.
```
> S = unicode:characters_to_binary("é", unicode, utf8).  
> {ok, tmpl} = erlydtl:compile_template("{{ s|striptags }}", tmpl).  
> {ok, IOList} = tmpl:render([{s, S}]).  
> byte_size(iolist_to_binary(IOList)).
1 %% would be 2 for utf-8
```

Basically, this filter and other simiar filters, such as `truncatechars`, are unusable for utf-8 encoded pages.

Also:
The `erlydtl_filters.erl` contains this code:
```
striptags(Input) when is_binary(Input) ->
    striptags(unicode:characters_to_list(Input));
```
Imo conversion to list from binary is redundant, `re` works with unicode just fine. 
Also `replace` calls could have `{return, binary}` in options instead of `{return, list}` - this should save some memory, but I'm not 100% sure, maybe in some other place this result is treated as list.

**Comment by kaos** [comment]
Yes, erlydtl is in need of some care and tender love to get proper unicode support.

I began some work on this a while back, but it wasn't finished. As it turns out, most of the filters need to be adapted, which is quite a lot of work, which haven't reached high enough on my prio list yet.. patches welcome ;)

**Comment by brigadier** [comment]
Do you have any idea on how it would be better to return result in a given encoding, not just `utf-8` or just `latin1`, as different projects have different encodings? I'm thinking about adding in and out encoding to `Options` of `compile_template/compile_file`, and then modify some filters to have optional `Options` too.  What do you think about this, is it possible at all?

**Comment by kaos** [comment]
That sounds like a good idea. Doable, but quite a bit of work, for sure.
